base/src/main/scala/typeclass/Bind.scala

```diff
@@ -3,7 +3,7 @@ package typeclass
 
 trait Bind[M[_]] {
   def apply: Apply[M]
-  def flatMap[A, B](ma: M[A])(f: A => M[B]): M[B]
+  def flatMap[A, B](ma: => M[A])(f: A => M[B]): M[B]
```

Comment:
This doesn't seem to be hugely beneficial; except for representable/distributive functors, we must always inspect `ma`.

It might be worth introducing something like ermine's [Suspendable](https://bitbucket.org/ermine-language/ermine-scala/src/40647cf0c50219e2a411a7e242fe16dc724d1e3c/core/src/main/scala/com/clarifi/reporting/Run.scala?at=default&fileviewer=file-view-default#Run.scala-8:11) that users could wrap around `ma` for non-strict `f`, e.g. [usage](https://bitbucket.org/ermine-language/ermine-writers/src/355667c10039839d71b076b420412106e0cd6e3d/writers/javafx/src/main/scala/com/clarifi/reporting/writers/JavaFXWriter.scala?at=default&fileviewer=file-view-default#JavaFXWriter.scala-143).